provisioner/aws-manager.js

```diff
@@ -1,33 +1,85 @@
 'use strict';
 
 var Promise = require('promise');
-var lodash = require('lodash');
 var assert = require('assert');
 var debug = require('debug')('aws-provisioner:aws-manager');
 var objFilter = require('../lib/objFilter');
 var shuffle = require('knuth-shuffle');
 var taskcluster = require('taskcluster-client');
+var base = require('taskcluster-base');
 
 /**
  * AWS EC2 state at a specific moment in time
  */
-function AwsManager(ec2, provisionerId, keyPrefix, pubKey, maxInstanceLife) {
+function AwsManager (ec2, provisionerId, keyPrefix, pubKey, maxInstanceLife, influx) {
   assert(ec2);
   assert(provisionerId);
   assert(keyPrefix);
   assert(pubKey);
   assert(maxInstanceLife);
+  assert(influx);
   this.ec2 = ec2;
   this.provisionerId = provisionerId;
   this.keyPrefix = keyPrefix;
   this.pubKey = pubKey;
   this.maxInstanceLife = maxInstanceLife;
+  this.influx = influx;
   this.__knownKeyPairs = [];
   this.__apiState = {
     instances: [],
     requests: [],
   };
   this.__internalState = [];
+
+  // This is a time series to measure how long it takes for instances to show up
+  // in the AWS api responses
+  this.Ec2ApiLagSeries = new base.stats.Series({
+    name: 'Ec2ApiLag',
+    columns: {
+      region: base.stats.types.String,
+      az: base.stats.types.String,
+      instanceType: base.stats.types.String,
+      workerType: base.stats.types.String,
+      id: base.stats.types.String,
+      // other columns should be obvious.
+      // This column is 0 for it showed up somehow, somewhere
+      // and 1 for being dropped on the floor
+      didShow: base.stats.types.Number,
+      // How many seconds to show up in API.  This is a maximum
+      // bound since we only check the API once every iteration
+      lag: base.stats.types.Number,
+    },
+  });
```

Comment:
cool.  i've done this.  I was trying to figure out how the reporter mechanism worked, and I thought I did what you listed above but it just didn't work.  I now have the reporter stuff working and I've verified that!